fixed a bug where we weren't able to properly process requests to the…

Jonahss opened the pull request:
… /status endpoint. related to appium #5335. Added tests. We should now properly proxy all commands that don't require a session id in the url

Jonahss committed:
fixed a bug where we weren't able to properly process requests to the /status endpoint. related to appium #5335. Added tests. We should now properly proxy all commands that don't require a session id in the url

moizjv commented:
+1 LGTM

sebv commented on lib/proxy.js:
```diff
@@ -52,44 +52,45 @@ class JWProxy {
 
-    if (requiresSessionId && this.sessionId === null) {
-      throw new Error("Trying to proxy a session command without session id");
+    const stripPrefixRe = new RegExp('^.+(/(session|status).*)$');
```
Better use regular regex syntax if the regex is not dynamic.

jlipps replied:
if the regex has `/` in it sometimes it's easier to read in this format

sebv commented on lib/proxy.js:
```diff
@@ -52,44 +52,45 @@ class JWProxy {
+      throw new Error("Trying to proxy a session command without session id");
     }
 
     const sessionBaseRe = new RegExp('^/session/([^/]+)');
```
Not your code but use shorter regex syntax.

sebv commented on lib/proxy.js:
```diff
@@ -52,44 +52,45 @@ class JWProxy {
     }
 
     const sessionBaseRe = new RegExp('^/session/([^/]+)');
-    if (sessionBaseRe.test(rest)) {
```
just use `match`
EDIT: or `replace`

sebv commented on lib/proxy.js:
```diff
@@ -52,44 +52,45 @@ class JWProxy {
+      remainingUrl = remainingUrl.replace(match[1], this.sessionId);
     } else if (requiresSessionId) {
-      throw new Error("Got bad session base with rest of url: " + rest);
+      throw new Error("Could not find :session section for url: " + remainingUrl);
```
This happens?

jlipps replied:
let's convert to interpolated string

sebv commented on test/proxy-specs.js:
```diff
@@ -53,6 +53,43 @@ describe('proxy', () => {
+  describe('getUrlForProxy', () => {
+    it('should modify session id, host, and port', async () => {
+      let j = mockProxy({sessionId: '123'});
+      let newUrl = j.getUrlForProxy('http://host.com:1234/wd/hub/session/456/element/200/value');
```
It's ok like this, but would just write:

```
mockProxy({sessionId: '123'})
   .getUrlForProxy('http://host.com:1234/wd/hub/session/456/element/200/value')
   .should.eql('http://localhost:4444/wd/hub/session/123/element/200/value'); 
```

or a small test helper wrapping the it method.

sebv commented:
lgtm, just some minor syntax issues.
best to wait for @jlipps comments.

jlipps commented:
yeah few comments, otherwise LGTM

Jonahss commented:
I added the string interpolation @jlipps suggested, fixed the tests as @sebv suggested. Is it ok if I leave the regex stuff for now? I would actually prefer to use less regex's since the bug this is fixing was only present cus the regex's kinda hid it.
